src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -8103,7 +8191,18 @@ void StorageReplicatedMergeTree::movePartitionToTable(const StoragePtr & dest_ta
 
     MergeTreeData & src_data = dest_table_storage->checkStructureAndGetMergeTreeData(*this, metadata_snapshot, dest_metadata_snapshot);
     auto src_data_id = src_data.getStorageID();
-    String partition_id = getPartitionIDFromQuery(partition, query_context);
+    String source_partition_id = getPartitionIDFromQuery(partition, query_context);
+
+    const auto dst_partition_expression = dest_metadata_snapshot->getPartitionKeyAST();
+    const auto src_partition_expression = metadata_snapshot->getPartitionKeyAST();
+    const auto is_partition_exp_the_same = queryToStringNullable(src_partition_expression) == queryToStringNullable(dst_partition_expression);
+
+    // TODO fix below
+    auto [destination_partition, destination_partition_id] = MergeTreePartitionCompatibilityVerifier::getDestinationPartitionAndPartitionId(
+        is_partition_exp_the_same,
+        src_data,
+        *dest_table_storage, {},
```

Comment:
I need the data parts to verify partitions are compatible and create destination partition id, but the parts are fetched later using drop range. Drop range depends on destination partition id.